lib/almond.js

```diff
@@ -281,6 +286,27 @@ module.exports = class Almond extends events.EventEmitter {
         };
     }
 
+    async setContext(context) {
+        if (context === null) {
+            this._context = {
+                timeout: Infinity,
+                code: 'null',
+                entities: {}
+            };
+        } else {
+            // timeout after 30 seconds
+            let timeout = Date.now() + 30000;
```

Comment:
Shouldn't the time be calculated on the server side?

Reply:
What server side? The NLP server is completely stateless & anonymous

Reply:
oh, never mind